src/client/tile.cpp

```diff
@@ -1784,6 +1781,46 @@ static void blit_with_interpolate_overlay(video::IImage *src, video::IImage *dst
 }
 
 /*
+	Apply color to destination
+*/
+static void apply_colorize(video::IImage *dst, v2s32 dst_pos, v2u32 size,
+		video::SColor color, int ratio, bool keep_alpha)
+{
+	u32 alpha = color.getAlpha();
+	video::SColor dst_c;
+	if (alpha == 0)
+		return;
+	if (((ratio == -1) && (alpha == 255)) || (ratio == 255)) {
```

Comment:
This might be a little less visually confusing if there were less ( (( )) )) parentheses.